### PR TITLE
SecureDrop Monitor, auto-update and notifications

### PR DESCRIPTION
**Pull request by katebee**
This adds a Python script intended to be run as a cron job on an EC2 instance. The instance needs to be running Tor since we can only access the site via Tor.

**N.B. Typically Tor listens for SOCKS connections on port 9050**

#### 👉 Send request to an onion site and validate response

The monitor script attempts to reach our onion site five times, with a 60 second interval between each attempt. The timeouts and sleep are quite generous, which gives the Tor network the benefit of the doubt. 🐢⏳

If all attempts fail, it will send notifications and update the bucket website config to point to the maintenance page. 🧟‍♀️💀

#### 👉 Fetch values from AWS Parameter Store

You can put parameters into parameter store as part of the CloudFormation, which is RAD:

Screenshot 2019-08-16 at 11 37 55

Using this will mean we only have to enter some parameters once - at the time of creating the stack in CloudFormation. We can then fetch the values in the Python script. We also fetch the onion address from AWS parameter store, so this can eventually be our **single source of truth** if we make other applications also fetch the value from the store. 🧙‍♀️✨

##### Pass notification

If everything is all okay Monitor will send a Hangouts Chat notification:

Screenshot 2019-08-16 at 11 09 17

The channel it messages is determined by the webhook URL that is stored in AWS parameter store. This parameter is conditional based on the `STAGE` of the application, therefore if you are running this as `DEV` you can avoid messaging the entire team... Which is also important if you are testing the fail / pass actions! 👩‍🔬🧪

##### Fail notification

If the healthcheck fails Monitor will send a different Hangouts Chat notification: 

Screenshot 2019-08-16 at 11 48 10

It will also send an email to an address stored in AWS parameter store:

Screenshot 2019-08-16 at 11 24 49

#### 👉 Control of the website configuration

The simplest approach to get this up and running is to have two different html documents in the S3 bucket. This script assumes they are called `index.html` and `maintenance.html`. The script will try to update the bucket website configuration each time it runs, setting it according to the outcome of the healthcheck. 👩‍⚕️🚑

### CODE REVIEW

File: monitor.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+import os
+import json
+from typing import Optional, Union, Dict
+
+import requests
+from boto3 import Session
+from botocore.exceptions import ClientError
+from requests.exceptions import RequestException
+
+METADATA_SERVICE = "http://169.254.169.254/latest/meta-data/"
```
Comment (itsibitzi) [inline monitor.py]: Is this being used?

Comment (katebee) [reply monitor.py]: Not anymore! Will remove 👍

File: requirements.txt [hunk 2]
```diff
@@ -4,7 +4,9 @@ docutils==0.14
 Jinja2==2.10.1
 jmespath==0.9.4
 MarkupSafe==1.1.1
+PySocks==1.7.0
```
Comment (itsibitzi) [inline requirements.txt]: Not uses as far as I can see?

**Comment by katebee** [comment]
@itsibitzi I am making the service shout in a dedicated chat channel:

Screenshot 2019-08-20 at 11 31 27

Will let me see how noisy it is! 😁 
The first false positive I expect is if the Tor service stops for some reason, or is slow to respond. 

@stiliyana-simeonova the outputs of the cronjob are currently going to `/tmp/listener.log` although they do need some improvement! I would like to switch to using the `logging` Python library and eventually ship logs off the box...

🚢 📃🚢 📃 

Thanks both!

**Pull request merged**